sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/models/UploadLogsOptions.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.ingestion.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.serializer.ObjectSerializer;
+
+/**
+ * The options model to configure the request to upload logs to Azure Monitor.
+ */
+@Fluent
+public final class UploadLogsOptions {
+    private ObjectSerializer objectSerializer;
+    private Integer maxConcurrency;
+
+    /**
+     * Returns the serializer to use to convert the log objects to JSON.
+     * @return the serializer to use to convert the log objects to JSON.
+     */
+    public ObjectSerializer getObjectSerializer() {
+        return objectSerializer;
+    }
+
+    /**
+     * Sets the serializer to use to convert the log objects to JSON.
+     * @param objectSerializer the serializer to use to convert the log objects to JSON.
+     * @return the update {@link UploadLogsOptions} instance.
+     */
+    public UploadLogsOptions setObjectSerializer(ObjectSerializer objectSerializer) {
```

Comment:
do we want to support non-json serialization for logs? if not, maybe `JsonSerializer` would be more precise?

Reply:
I don't know if the service will ever support any other formats like avro in the future. But as of now, it's only JSON. So, we could just use JsonSerializer. I will check with the service team and update this based on their feedback for preview 2.